staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured_test.go

```diff
@@ -34,3 +34,17 @@ func TestUnstructuredList(t *testing.T) {
 		t.Fatalf("unexpected fields: %#v", items[0])
 	}
 }
+
+func TestNilDeletionTimestamp(t *testing.T) {
+	var u Unstructured
+	u.SetDeletionTimestamp(u.GetDeletionTimestamp())
+	del := u.GetDeletionTimestamp()
+	if del != nil {
+		t.Errorf("unexpected non-nil deletion timestamp: %v", del)
+	}
```

Comment:
I've added an extra check before `SetDeletionTimestamp()` is invoked if that is what you wanted? We need to check that getter returns `nil` after setter is invoked so I want to have both checks there.